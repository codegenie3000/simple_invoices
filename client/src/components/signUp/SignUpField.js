import React from 'react';

import globalStyles from '../../stylesheets/GlobalElements.module.css';
import styles from './SignUpField.module.css';

// Presentational Component - accepts props
const SurveyField = ({ input, name, type, label, required, meta: { touched, error, warning } }) => {
    return (
        <div className={ styles.formWidth }>
            <label className={ globalStyles.styledLabel }>{label}</label>
            <input {...input} name={name} required={required} type={type} className={ globalStyles.styledForm }/>
            <div>
                {touched && ((error && <span className={globalStyles.formWarning}>{error}</span>) || (warning && <span className={globalStyles.formWarning}>{warning}</span>))}
            </div>
        </div>
    );
};

export default SurveyField;