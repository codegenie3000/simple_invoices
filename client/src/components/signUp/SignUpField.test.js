import React from 'react';
import { shallow } from 'enzyme';

import SignUpField from './SignUpField';

describe('<SignUpField />', () => {
    let component;
    let testName = 'testName';
    let testLabel = 'testLabel';
    let testRequired = true;

    beforeEach(() => {
        component = shallow(<SignUpField
            name={testName}
            label={testLabel}
            required={testRequired}
            meta={
                {
                    touched: '',
                    error: '',
                    warning: ''
                }
            }
        />);
    });
    it('matches snapshot', () => {
        expect(component).toMatchSnapshot();
    });
    it('tests if props are working properly', () => {
        expect(component.at(0).children().at(1).props().name).toEqual(testName);
        expect(component.at(0).children().at(0).children().at(0).text()).toEqual(testLabel);
        expect(component.at(0).children().at(1).props().required).toEqual(testRequired);
    });
});