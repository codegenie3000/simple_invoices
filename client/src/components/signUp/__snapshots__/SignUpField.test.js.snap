// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignUpField /> matches snapshot 1`] = `
<div
  className="formWidth"
>
  <label
    className="styledLabel"
  >
    testLabel
  </label>
  <input
    className="styledForm"
    name="testName"
    required={true}
  />
  <div />
</div>
`;
